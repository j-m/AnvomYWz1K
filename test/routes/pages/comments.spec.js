'use strict'

const request = require('supertest')

process.env.DATABASE = ':memory:'
process.env.REVIEWS_PER_PAGE = '2'

const app = require('../../../src/app/koa')
const connection = require('../../../src/database/connection')
const scrubSnapshot = require('../../../src/test/util/scrubSnapshot')

beforeAll(async() => {
  await connection.open()
  await Promise.all([
    connection.run('insert.game', ...[
      'gameID',
      'steamAppId',
      'title',
      'summary', null, null, null, null, null, null, null, null
    ]),
    connection.run('insert.member', 'any', 'userID', 'any'),
  ])
  await connection.run('insert.review', 'gameID', 'userID', '1', 'review body', 'short')
  await connection.run('insert.comment', 'gameID', 'userID', 'short', 'userID', 'comment')
})

afterAll(async() => {
  await connection.close()
})

describe('routes get game', () => {
  test('should contain templated text with generated thumbnail, banner, and store', async() => {
    const response = await request(app.callback()).get('/games/gameID/userID/short')
    expect(response.status).toEqual(200)
    expect(scrubSnapshot(response.text)).toMatchSnapshot()
  })

})
