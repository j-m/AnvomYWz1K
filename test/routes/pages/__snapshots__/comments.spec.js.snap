// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes get game should contain templated text with generated thumbnail, banner, and store 1`] = `
"<!DOCTYPE html>

<html lang=\\"en\\">
	<head>
		<meta charset=\\"utf-8\\">
		<title>Game Reviews - Comments -  -  - </title>
    <link href=\\"https://fonts.googleapis.com/css?family=Roboto&display=swap\\" rel=\\"stylesheet\\">
    
    <link href=\\"../../../css/global.css\\" type=\\"text/css\\" rel=\\"stylesheet\\" />
    <script src=\\"../../../js/global.js\\" type=\\"text/javascript\\"></script>
    
		<link href=\\"../../../css/loginRegistration.css\\" type=\\"text/css\\" rel=\\"stylesheet\\" />
    <script src=\\"../../../js/loginRegistration.js\\" type=\\"text/javascript\\"></script>
    
    <link href=\\"../../../css/review.css\\" type=\\"text/css\\" rel=\\"stylesheet\\" />
    
    <link href=\\"../../../css/comments.css\\" type=\\"text/css\\" rel=\\"stylesheet\\" />
    <script src=\\"../../../js/comments.js\\" type=\\"text/javascript\\"></script>
    
    <script src=\\"https://cdn.rawgit.com/showdownjs/showdown/1.9.1/dist/showdown.min.js\\" type=\\"text/javascript\\"></script>
	</head>
  <body onload=\\"parseDifferent(\`review body\`)\\">
    <div id=\\"welcome\\">
      <p>Welcome, guest! <a href=\\"#\\" onclick=\\"show('login')\\">Login</a> or <a href=\\"#\\" onclick=\\"show('register')\\">Register</a> </p>
    </div>
    
    <div class=\\"popup\\" onclick=\\"hide('login', event)\\">
      <form id=\\"login\\" onsubmit=\\"login(event)\\">
        <div class=\\"input_header\\">
        	<label for=\\"username\\">Username</label>
          <p class=\\"error USERNAME_MISSING\\">This field is required. </p>
        </div>
        <input type=\\"text\\" id=\\"username\\" name=\\"username\\" autocomplete=\\"current-username\\" required/>
        <p class=\\"error USERNAME_UNKNOWN\\">This username does not exist. </p>
        
        <div class=\\"input_header\\">
        	<label for=\\"password\\">Password</label>
          <p class=\\"error PASSWORD_MISSING\\">This field is required. </p>
        </div>
        <input type=\\"password\\" id=\\"password\\" name=\\"password\\" autocomplete=\\"current-password\\" required/>
        <p class=\\"error PASSWORD_INCORRECT\\">Incorrect password. </p>
        
      	<input type=\\"submit\\" value=\\"Log in\\"/>
        <a href=\\"#forgot\\">Forgot password</a>
      </form>
    </div>
    
    <div class=\\"popup\\" onclick=\\"hide('register', event)\\">
      <form id=\\"register\\" onsubmit=\\"register(event)\\">
        <div class=\\"input_header\\">
        	<label for=\\"email\\">Email</label>
          <p class=\\"error EMAIL_MISSING\\">This field is required. </p>
        </div>
        <input type=\\"email\\" id=\\"email\\" name=\\"email\\" autocomplete=\\"email\\" required/>
        
        <div class=\\"input_header\\">
        	<label for=\\"new-username\\">Username</label>
          <p class=\\"error USERNAME_MISSING\\">This field is required. </p>
          <p class=\\"error USERNAME_AVAILABLE\\">Username available. </p>
        </div>
        <input type=\\"text\\" id=\\"new-username\\" name=\\"new-username\\" autocomplete=\\"none\\" required/>
        <button type=\\"button\\" onclick=\\"checkUsernameAvailability()\\">&#x21bb;</button>
        <p class=\\"error USERNAME_IN_USE\\">This username is already in use. </p>
        <p class=\\"info USERNAME_BAD_REGEX\\">Usernames can only contain letters and numbers. <br/> They may contain, but not start or end with, hyphens(-). </p>
        
        <div class=\\"input_header\\">
        	<label for=\\"new-password\\">Password</label>
          <p class=\\"error PASSWORD_MISSING\\">This field is required. </p>
        </div>
        <input type=\\"password\\" id=\\"new-password\\" name=\\"new-password\\" minlength=\\"10\\" autocomplete=\\"new-password\\" required/>
        <p class=\\"info PASSWORD_TOO_SHORT\\">Passwords must be at least 10 characters long. </p>
        
      	<input type=\\"submit\\" value=\\"Register\\">
      </form>
    </div>
    
    <div id=\\"review\\">
      <div class=\\"header\\">
        <p class=\\"rating\\">1</p>
        <p class=\\"author\\">by <a href=\\"/members/\\"></a></p>
        <div class=\\"dates\\">
          <p class=\\"posted\\">Posted: <time datetime=\\"timestamp\\">timestamp</time></p>
        </div>
        <div class=\\"buttons\\">
          <button class=\\"visibility\\">V</button>
          <button class=\\"flag\\">F</button>
          <button class=\\"edit\\">E</button>
        </div>
      </div>
      <hr/>
      <div class=\\"middle\\">
        <div class=\\"votes\\">
          <button class=\\"upvote\\"></button>
          <p class=\\"score\\">0</p>
          <button class=\\"downvote\\"></button>
        </div>
        <p id=\\"reviewBody\\">review body</p>
      </div>
      <hr/>
      <div id=\\"comments\\">
        <div class=\\"comment\\">
          <time class=\\"posted\\" datetime=\\"timestamp\\">timestamp</time>
          <span class=\\"author\\">userID:</span>
          <span class=\\"commentBody\\">comment</span>
        </div>
        <hr/>
      </div>
      <form id=\\"commentForm\\" onsubmit=\\"somethingDifferent(event, '', '', '')\\">
        <input type=\\"text\\" id=\\"commentInput\\"name=\\"comment\\" autocomplete=\\"comment\\" required/>
        <input type=\\"submit\\" value=\\"Comment\\"/>
      </form>
    </div>
  </body>

</html>
"
`;
